lib/sisimai/bite/email/google.rb

```diff
@@ -0,0 +1,320 @@
+            # ed was: 554 554 5.7.0 Header error (state 18).
+            #
+            # -- OR --
+            #
+            # Technical details of permanent failure:=20
+            # Google tried to deliver your message, but it was rejected by the server for=
+            # the recipient domain example.jp by mx.example.jp. [192.0.2.49].
+            #
+            # The error that the other server returned was:
+            # 550 5.1.1 <userunknown@example.jp>... User Unknown
+            #
+            v = dscontents[-1]
+
+            if cv = e.match(/\A[ \t]+([^ ]+[@][^ ]+)\z/)
+              # kijitora@example.jp: 550 5.2.2 <kijitora@example>... Mailbox Full
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+
+              addr0 = Sisimai::Address.s3s4(cv[1])
+              if Sisimai::RFC5322.is_emailaddress(addr0)
+                v['recipient'] = addr0
+                recipients += 1
+              end
+
+            else
+              v['diagnosis'] ||= ''
+              v['diagnosis']  += e + ' '
+            end
+          end
+        end
+
+        return nil if recipients.zero?
+        require 'sisimai/string'
+        require 'sisimai/smtp/status'
+
+        dscontents.map do |e|
+          e['agent']     = self.smtpagent
+          e['diagnosis'] = Sisimai::String.sweep(e['diagnosis'])
+
+          unless e['rhost']
+            # Get the value of remote host
+            if cv = e['diagnosis'].match(/[ \t]+by[ \t]+([^ ]+)[.][ \t]+\[(\d+[.]\d+[.]\d+[.]\d+)\][.]/)
+              # Google tried to deliver your message, but it was rejected by
+              # the server for the recipient domain example.jp by mx.example.jp. [192.0.2.153].
+              hostname = cv[1]
+              ipv4addr = cv[2]
+              if hostname =~ /[-0-9a-zA-Z]+[.][a-zA-Z]+\z/
+                # Maybe valid hostname
+                e['rhost'] = hostname.downcase
+              else
+                # Use IP address instead
+                e['rhost'] = ipv4addr
+              end
+            end
+          end
+
+          if cv = e['diagnosis'].match(/[(]state[ ](\d+)[)][.]/)
+            statecode0 = cv[1]
+          end
+          if StateTable[statecode0]
+            # (state *)
+            e['reason']  = StateTable[statecode0]['reason']
+            e['command'] = StateTable[statecode0]['command']
+
+          else
+            # No state code
+            ReFailure.each_key do |r|
+              # Verify each regular expression of session errors
+              next unless e['diagnosis'] =~ ReFailure[r]
+              e['reason'] = r.to_s
+              break
+            end
+          end
+          e['status'] = Sisimai::SMTP::Status.find(e['diagnosis'])
+
+          if e['reason']
```

Comment:
__[rubocop]__
* Style/Next: Use `next` to skip iteration. [:ref](https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)

<!-- Comment by [SideCI](https://sideci.com) -->